biz.aQute.repository/src/aQute/bnd/jpm/Repository.java

```diff
@@ -1606,6 +1606,7 @@ protected void fireBundleAdded(File file) throws IOException {
 	public void setRegistry(Registry registry) {
 		this.registry = registry;
 		this.httpClient = registry.getPlugin(HttpClient.class);
+		this.httpClient.setRegistry(registry);
```

Comment:
This seems like a threading issue. The getPlugin returns the one? HttpClient object and then the caller set a registry in it. What happens when there are multiple callers to get the HttpClient plugin? Either on this thread or across multiple threads?